packages/dev/core/src/Materials/clipPlaneMaterialHelper.ts

```diff
@@ -28,18 +28,49 @@ export function addClipPlaneUniforms(uniforms: string[]): void {
 /** @internal */
 export function prepareDefinesForClipPlanes(primaryHolder: IClipPlanesHolder, secondaryHolder: IClipPlanesHolder, defines: string[] | Record<string, any>): boolean {
     let changed = false;
-    let clipPlane = primaryHolder.clipPlane ?? secondaryHolder.clipPlane;
-    changed = addDefine(clipPlane, defines, "CLIPPLANE", "#define CLIPPLANE") || changed;
-    clipPlane = primaryHolder.clipPlane2 ?? secondaryHolder.clipPlane2;
-    changed = addDefine(clipPlane, defines, "CLIPPLANE2", "#define CLIPPLANE2") || changed;
-    clipPlane = primaryHolder.clipPlane3 ?? secondaryHolder.clipPlane3;
-    changed = addDefine(clipPlane, defines, "CLIPPLANE3", "#define CLIPPLANE3") || changed;
-    clipPlane = primaryHolder.clipPlane4 ?? secondaryHolder.clipPlane4;
-    changed = addDefine(clipPlane, defines, "CLIPPLANE4", "#define CLIPPLANE4") || changed;
-    clipPlane = primaryHolder.clipPlane5 ?? secondaryHolder.clipPlane5;
-    changed = addDefine(clipPlane, defines, "CLIPPLANE5", "#define CLIPPLANE5") || changed;
-    clipPlane = primaryHolder.clipPlane6 ?? secondaryHolder.clipPlane6;
-    changed = addDefine(clipPlane, defines, "CLIPPLANE6", "#define CLIPPLANE6") || changed;
+
+    const clipPlane = !!(primaryHolder.clipPlane ?? secondaryHolder.clipPlane);
+    const clipPlane2 = !!(primaryHolder.clipPlane2 ?? secondaryHolder.clipPlane2);
+    const clipPlane3 = !!(primaryHolder.clipPlane3 ?? secondaryHolder.clipPlane3);
+    const clipPlane4 = !!(primaryHolder.clipPlane4 ?? secondaryHolder.clipPlane4);
+    const clipPlane5 = !!(primaryHolder.clipPlane5 ?? secondaryHolder.clipPlane5);
+    const clipPlane6 = !!(primaryHolder.clipPlane6 ?? secondaryHolder.clipPlane6);
+
+    if (Array.isArray(defines)) {
+        changed = addDefineString(clipPlane, defines, "#define CLIPPLANE") || changed;
+        changed = addDefineString(clipPlane2, defines, "#define CLIPPLANE2") || changed;
+        changed = addDefineString(clipPlane3, defines, "#define CLIPPLANE3") || changed;
+        changed = addDefineString(clipPlane4, defines, "#define CLIPPLANE4") || changed;
+        changed = addDefineString(clipPlane5, defines, "#define CLIPPLANE5") || changed;
+        changed = addDefineString(clipPlane6, defines, "#define CLIPPLANE6") || changed;

```

Comment:
Actually, we can also "linearize" the code here and simply do (that's what each sub-system was doing before the factoring):
```javascript
if (clipPlane) defines.push("#define CLIPPLANE");
if (clipPlane2) defines.push("#define CLIPPLANE2");
etc
```
because when `prepareDefinesForClipPlanes` is called with a string array, the array is always initialized empty beforehand (I have checked all occurrences of `prepareDefinesForClipPlanes` in the code base). Also, we don't need to return a flag true/false in this case, because the caller does not care and does not test it: it will do a `defines.join("\n")` and check the result with the previous stored value.